semaphore/integration_tests.py

```diff
@@ -0,0 +1,119 @@
+#/usr/bin/env python
+from datetime import date, time, datetime, timedelta
+from random import randrange, getrandbits
+import json
+import requests
+
+SPLUNK_HOST = "https://mbta.splunkcloud.com:8089"
+SAVED_SEARCH = "Top 10 trip plans %2B 10 latest trip plans"
+OTP_DEV = "https://dev.otp.mbtace.com"
+OTP_PROD = "https://prod.otp.mbtace.com"
+
+PREDEFINED_PLANS = [
+    # Airport shuttles 1
+    {"Terminal E - Arrivals Level::42.369344,-71.020238", "Wood Island::42.37964,-71.022865"},
+
+    # Airport shuttles 2
+    {"Assembly::42.392811,-71.077257", "Terminal C - Departures Level::42.366635,-71.017167"},
+
+    # Connection between bus routes in the middle of the routes
+    {"Lexington St @ Willow St::42.4732,-71.17253", "Main St, Woburn, MA, USA::42.4928705,-71.1544787"},
+
+    # Logan express
+    {"Back Bay::42.34735,-71.075727", "Logan International Airport, Boston, MA, USA::42.3658907,-71.017547"}
+]
+
+
+def splunk_json_request(path):
+    return requests.get(f"{SPLUNK_HOST}{path}",
+                        params={"output_mode": "json"},
+                        auth=("lboyarsky@mbta.com", "XXX"),
+                        verify=False).json()
+
+
+def splunk_get_search_name():
+    data = splunk_json_request(f"/services/saved/searches/{SAVED_SEARCH}/history")
+
+    if "entry" not in data or len(data.get("entry")) < 1 or "name" not in data.get("entry")[0]:
+        raise RuntimeError("Unable to get results reference from splunk")
+
+    return data.get("entry")[0].get("name")
+
+
+def splunk_get_search_results(name):
+    data = splunk_json_request(f"/services/search/jobs/{name}/results")
+
+    if "results" not in data or len(data.get("results")) <= 10:
+        raise RuntimeError("Unable to get trip data from splunk")
+
+    return [{r.get("fromPlace"), r.get("toPlace")} for r in data.get("results")]
+
+
+def get_testing_dates():
+    # returns random times on next Wednesday and next Sunday
+
+    tomorrow = (
+        datetime.combine(date.today() + timedelta(days=1), time(randrange(6, 21), randrange(0, 59)))
+    )
+
+    return {tomorrow + timedelta(days=(2 - tomorrow.weekday()) % 7),
+            tomorrow + timedelta(days=(6 - tomorrow.weekday()) % 7)}
+
+
+def get_trip_plans(environment, from_place, to_place, trip_date, arrive_by):
+    url = f"{(OTP_PROD if environment == 'prod' else OTP_DEV)}/otp/routers/default/plan"
+
+    params = {
+        "fromPlace": from_place,
+        "toPlace": to_place,
+        "date": trip_date.strftime("%Y-%m-%d"),
+        "time": trip_date.strftime("%I:%M%p"),
+        "showIntermediateStops": "true",
+        "format": "json",
+        "locale": "en",
+        "mode": "TRAM,SUBWAY,FERRY,RAIL,BUS,WALK",
+        "walkReluctance": "5",
+        "arriveBy": "true" if arrive_by else "false",
+    }
+
+    return requests.get(url, params).json()
+
+
+def compare_plans(plan1, plan2):
```

Comment:
do you think you could prove an example of what a typical argument would look like?

Reply:
Do you mean an example of a trip plan being compared? Here you go:

```json
{
   "debugOutput":{
      "pathCalculationTime":48,
      "pathTimes":[
         7,
         13,
         28
      ],
      "precalculationTime":36,
      "renderingTime":0,
      "timedOut":false,
      "totalTime":84
   },
   "elevationMetadata":{
      "ellipsoidToGeoidDifference":-28.142313888019384,
      "geoidElevation":false
   },
   "plan":{
      "date":1563742380000,
      "from":{
         "lat":42.352271,
         "lon":-71.055242,
         "name":"South Station",
         "orig":"South Station",
         "vertexType":"NORMAL"
      },
      "itineraries":[
         {
            "duration":2844,
            "elevationGained":0.0,
            "elevationLost":0.0,
            "endTime":1563742380000,
            "legs":[
               {
                  "agencyTimeZoneOffset":-14400000,
                  "arrivalDelay":0,
                  "departureDelay":0,
                  "distance":67.29053121698426,
                  "duration":263.0,
                  "endTime":1563739800000,
                  "from":{
                     "departure":1563739537000,
                     "lat":42.352271,
                     "lon":-71.055242,
                     "name":"South Station",
                     "orig":"South Station",
                     "vertexType":"NORMAL"
                  },
                  "interlineWithPreviousLeg":false,
                  "intermediateStops":[

                  ],
                  "legGeometry":{
                     "length":2,
                     "points":"u|naGh~tpLdByA"
                  },
                  "mode":"WALK",
                  "pathway":false,
                  "realTime":false,
                  "rentedBike":false,
                  "route":"",
                  "startTime":1563739537000,
                  "steps":[
                     {
                        "absoluteDirection":"SOUTHEAST",
                        "area":false,
                        "bogusName":false,
                        "distance":67.29053121698426,
                        "elevation":[

                        ],
                        "lat":42.352271,
                        "lon":-71.055242,
                        "relativeDirection":"DEPART",
                        "stayOn":false,
                        "streetName":"Transfer"
                     }
                  ],
                  "to":{
                     "arrival":1563739800000,
                     "departure":1563739800000,
                     "lat":42.351763,
                     "lon":-71.054797,
                     "name":"South Station",
                     "stopId":"1:South Station",
                     "stopIndex":0,
                     "stopSequence":1,
                     "vertexType":"TRANSIT",
                     "zoneId":"CR-zone-1A"
                  },
                  "transitLeg":false
               },
               {
                  "agencyId":"1",
                  "agencyName":"MBTA",
                  "agencyTimeZoneOffset":-14400000,
                  "agencyUrl":"http://www.mbta.com",
                  "alightRule":"coordinateWithDriver",
                  "arrivalDelay":0,
                  "departureDelay":0,
                  "distance":12027.831532914552,
                  "duration":780.0,
                  "endTime":1563740580000,
                  "from":{
                     "arrival":1563739800000,
                     "departure":1563739800000,
                     "lat":42.351763,
                     "lon":-71.054797,
                     "name":"South Station",
                     "stopId":"1:South Station",
                     "stopIndex":0,
                     "stopSequence":1,
                     "vertexType":"TRANSIT",
                     "zoneId":"CR-zone-1A"
                  },
                  "headsign":"Kingston",
                  "interlineWithPreviousLeg":false,
                  "intermediateStops":[
                     {
                        "arrival":1563740160000,
                        "departure":1563740160000,
                        "lat":42.321079,
                        "lon":-71.052155,
                        "name":"JFK/UMass",
                        "stopId":"1:JFK/UMASS",
                        "stopIndex":1,
                        "stopSequence":2,
                        "vertexType":"TRANSIT",
                        "zoneId":"CR-zone-1A"
                     }
                  ],
                  "legGeometry":{
                     "length":231,
                     "points":"innaG`~tpLzAp@xAz@vAx@xAz@xAz@xAz@tA~@tA`AxAz@xAz@vAz@xAx@vA~@xAx@zAr@|Aj@`BRbBF`B?bBCbBCbBEbBC`BEbBEbBCbBD`B@bBBbBAbBI`BU`B[~AWbBMbBK`BEbBGbBGbBE`BCbBA`BM`Ba@zAk@xAy@rAgAlAwAhAwAhA{AdA}AfA}AdA}AfA}AdA}AfA_BhAyApAiAvA_A|Ak@`BY`BQbBOPAnAMbBK`BKbBEbBCbBA`BAbB?bBCbBG`BF`BLbB?bBN`BX~A`@|Af@~Ad@~Af@~Ad@|Af@|Ah@~Ad@~Af@~Ab@~AZ`BZ`BX`BV`BX`BV`BT`BV`B\\~A\\`BTbBNbBB`BEbBU~Ac@zAo@xA{@rAgAlAoAlAuAjAuAlAuAnAoAnAoAjAuAjAuAlAuAlAsAjAuAjAwAjAuAlAuAjAuAjAwAjAuAjAuAlAuAjAwAjAuAjAuAlAuAjAuAjAuAlAuAjAuAjAwAjAwAjAwAjAuAjAuAlAuAlAqAlAsAlAqAlAuAjAuAlAuAjAuAjAwAjAuAjAwAjAuAlAuAjAwAhAwAlAsArAgAtAcAzAq@~Ag@`B]~Ac@~Ae@|Ak@xAu@rAgAnAoApAoAnAoAlAoAnAqAnAqAlAsAjAuAjAuAlAwAjAuAjAuAjAuAnAqAlAqAnAqAnAqAlAqAlAuAlAsAlAsAlAsAlAsAlAsAlAsAlAsAlAsAjAuAnAqAnAoAnAmAnAqAjAuAnAqAlAsAlAqAnAqAlAuAjAsAlAsAnAqAnAoAnAqAlAqAlAsAlAsAnAqAlAsAjAsAnAsAlAqAnAqAlAqAnAsAlAsAjAsAlAsAlAuAnAoApAkAvA}@|Ai@xA{@RM"
                  },
                  "mode":"RAIL",
                  "pathway":false,
                  "realTime":false,
                  "rentedBike":false,
                  "route":"Kingston/Plymouth Line",
                  "routeColor":"80276C",
                  "routeId":"1:CR-Kingston",
                  "routeLongName":"Kingston/Plymouth Line",
                  "routeTextColor":"FFFFFF",
                  "routeType":2,
                  "serviceDate":"20190721",
                  "startTime":1563739800000,
                  "steps":[

                  ],
                  "to":{
                     "arrival":1563740580000,
                     "lat":42.251809,
                     "lon":-71.005409,
                     "name":"Quincy Center",
                     "orig":"Quincy Center",
                     "stopId":"1:Quincy Center",
                     "stopIndex":2,
                     "stopSequence":3,
                     "vertexType":"TRANSIT",
                     "zoneId":"CR-zone-1A-1"
                  },
                  "transitLeg":true,
                  "tripId":"1:CR-Sunday-Spring-19-2037",
                  "tripShortName":"2037"
               }
            ],
            "startTime":1563739536000,
            "tooSloped":false,
            "transfers":0,
            "transitTime":780,
            "waitingTime":1801,
            "walkDistance":0.0,
            "walkLimitExceeded":false,
            "walkTime":263
         },
         {
            "duration":3139,
            "elevationGained":0.0,
            "elevationLost":0.0,
            "endTime":1563742380000,
            "legs":[
               {
                  "agencyTimeZoneOffset":-14400000,
                  "arrivalDelay":0,
                  "departureDelay":0,
                  "distance":0.0,
                  "duration":78.0,
                  "endTime":1563739320000,
                  "from":{
                     "departure":1563739242000,
                     "lat":42.352271,
                     "lon":-71.055242,
                     "name":"South Station",
                     "orig":"South Station",
                     "vertexType":"NORMAL"
                  },
                  "interlineWithPreviousLeg":false,
                  "intermediateStops":[

                  ],
                  "legGeometry":{
                     "length":2,
                     "points":"u|naGh~tpL??"
                  },
                  "mode":"WALK",
                  "pathway":false,
                  "realTime":false,
                  "rentedBike":false,
                  "route":"",
                  "startTime":1563739242000,
                  "steps":[
                     {
                        "absoluteDirection":"SOUTH",
                        "area":false,
                        "bogusName":false,
                        "distance":0.0,
                        "elevation":[

                        ],
                        "lat":42.352271,
                        "lon":-71.055242,
                        "relativeDirection":"DEPART",
                        "stayOn":false,
                        "streetName":"Transfer"
                     }
                  ],
                  "to":{
                     "arrival":1563739320000,
                     "departure":1563739320000,
                     "lat":42.352271,
                     "lon":-71.055242,
                     "name":"South Station",
                     "stopCode":"70079",
                     "stopId":"1:70079",
                     "stopIndex":9,
                     "stopSequence":90,
                     "vertexType":"TRANSIT"
                  },
                  "transitLeg":false
               },
               {
                  "agencyId":"1",
                  "agencyName":"MBTA",
                  "agencyTimeZoneOffset":-14400000,
                  "agencyUrl":"http://www.mbta.com",
                  "alerts":[
                     {
                        "alertDescriptionText":"As part of the Wollaston Station reconstruction project, passengers may ride the Commuter Rail in both directions between Quincy Center and South Station with a CharlieCard or CharlieTicket.",
                        "effectiveStartDate":1560746449000
                     }
                  ],
                  "arrivalDelay":0,
                  "departureDelay":0,
                  "distance":8994.708687246453,
                  "duration":960.0,
                  "endTime":1563740280000,
                  "from":{
                     "arrival":1563739320000,
                     "departure":1563739320000,
                     "lat":42.352271,
                     "lon":-71.055242,
                     "name":"South Station",
                     "stopCode":"70079",
                     "stopId":"1:70079",
                     "stopIndex":9,
                     "stopSequence":90,
                     "vertexType":"TRANSIT"
                  },
                  "headsign":"Braintree",
                  "interlineWithPreviousLeg":false,
                  "intermediateStops":[
                     {
                        "arrival":1563739500000,
                        "departure":1563739500000,
                        "lat":42.342622,
                        "lon":-71.056967,
                        "name":"Broadway",
                        "stopCode":"70081",
                        "stopId":"1:70081",
                        "stopIndex":10,
                        "stopSequence":100,
                        "vertexType":"TRANSIT"
                     },
                     {
                        "arrival":1563739620000,
                        "departure":1563739620000,
                        "lat":42.330154,
                        "lon":-71.057655,
                        "name":"Andrew",
                        "platformCode":"1",
                        "stopCode":"70083",
                        "stopId":"1:70083",
                        "stopIndex":11,
                        "stopSequence":110,
                        "vertexType":"TRANSIT"
                     },
                     {
                        "arrival":1563739800000,
                        "departure":1563739800000,
                        "lat":42.320685,
                        "lon":-71.052391,
                        "name":"JFK/UMass",
                        "stopCode":"70095",
                        "stopId":"1:70095",
                        "stopIndex":12,
                        "stopSequence":120,
                        "vertexType":"TRANSIT"
                     }
                  ],
                  "legGeometry":{
                     "length":124,
                     "points":"g~naGd}tpLnDmM^i@POl@UhAQj@Gb@?r@HZFj@Pz@b@fe@h[pCAzlAm@f@?j@TxF|BpBp@\\NNBV?|@[j@[l@g@~B}CTYn@kA~A}BZg@hBeDnAkBz@gAh@k@j@g@fAs@dAe@~@YpCc@l@GRERC\\G`AQ^C@?JA|AGf@A^EjCGrEIz@AdEAvHY|BB`ALjB\\fF~AdDbA|IpCxCv@xD|@rWdEXFrB`@rANvAHz@ChBIx@K`AU|@_@hAi@HElA{@hA{@zHuI~DwE`DqDpLkNhC{ClEiFhLaN`@a@h@q@RURU`DqDdAiA~@gAx@_AHKHK~E{FdBqBfAgAdDyDxDoE~DeF|BkClAuANQBELMHM|@eAVYvAoA`As@dAs@jAm@l@QpAa@xDcAfA]x@Yj@Wv@a@`@YhAaAhCeCz@aAtBwB"
                  },
                  "mode":"SUBWAY",
                  "pathway":false,
                  "realTime":false,
                  "rentedBike":false,
                  "route":"Red Line",
                  "routeColor":"DA291C",
                  "routeId":"1:Red",
                  "routeLongName":"Red Line",
                  "routeTextColor":"FFFFFF",
                  "routeType":1,
                  "serviceDate":"20190721",
                  "startTime":1563739320000,
                  "steps":[

                  ],
                  "to":{
                     "arrival":1563740280000,
                     "departure":1563740412000,
                     "lat":42.275275,
                     "lon":-71.029583,
                     "name":"North Quincy",
                     "platformCode":"2",
                     "stopCode":"70097",
                     "stopId":"1:70097",
                     "stopIndex":13,
                     "stopSequence":180,
                     "vertexType":"TRANSIT"
                  },
                  "transitLeg":true,
                  "tripId":"1:40525673-20:45-BraintreeNQuincyL"
               },
               {
                  "agencyTimeZoneOffset":-14400000,
                  "alerts":[
                     {
                        "alertDescriptionText":"During garage construction, 592 spots at the North Quincy Hancock Street lot will be unavailable. More than 600 spots will be available in two surface lots: \r\n350 spots at Newport Avenue lot \r\n260 spots at Hancock Street lot \r\n\r\nOn weekdays, the Hancock Street lot typically fills by about 7:30 AM, and the Newport Avenue lot fills between 7:45 & 8 AM.\r\n\r\nAlternative parking and travel options include: \r\n403 spots at Wollaston Station during the station closure, with free shuttle service between North Quincy, Quincy Center, and Wollaston Stations.\r\n\r\nFor more information visit: www.mbta.com/northquincy",
                        "alertUrl":"https://www.mbta.com/projects/north-quincy-station-garage",
                        "effectiveStartDate":1550395800000
                     },
                     {
                        "alertDescriptionText":"At the street level, travel across the parking lot to the south lobby/busway. At the south lobby, use North Quincy Elevator 900 to reach the platform.",
                        "effectiveStartDate":1563367285000
                     }
                  ],
                  "arrivalDelay":0,
                  "departureDelay":0,
                  "distance":0.0,
                  "duration":108.0,
                  "endTime":1563740520000,
                  "from":{
                     "arrival":1563740280000,
                     "departure":1563740412000,
                     "lat":42.275275,
                     "lon":-71.029583,
                     "name":"North Quincy",
                     "platformCode":"2",
                     "stopCode":"70097",
                     "stopId":"1:70097",
                     "stopIndex":13,
                     "stopSequence":180,
                     "vertexType":"TRANSIT"
                  },
                  "interlineWithPreviousLeg":false,
                  "intermediateStops":[

                  ],
                  "legGeometry":{
                     "length":2,
                     "points":"m{_aG|}opLuHjD"
                  },
                  "mode":"WALK",
                  "pathway":true,
                  "realTime":false,
                  "rentedBike":false,
                  "route":"",
                  "startTime":1563740412000,
                  "steps":[
                     {
                        "absoluteDirection":"NORTH",
                        "area":false,
                        "bogusName":false,
                        "distance":0.0,
                        "elevation":[

                        ],
                        "lat":42.275275,
                        "lon":-71.029583,
                        "relativeDirection":"DEPART",
                        "stayOn":false,
                        "streetName":"pathway"
                     }
                  ],
                  "to":{
                     "arrival":1563740520000,
                     "departure":1563740520000,
                     "lat":42.276824,
                     "lon":-71.030448,
                     "name":"North Quincy",
                     "stopCode":"9070098",
                     "stopId":"1:9070098",
                     "stopIndex":0,
                     "stopSequence":1,
                     "vertexType":"TRANSIT"
                  },
                  "transitLeg":false
               },
               {
                  "agencyId":"1",
                  "agencyName":"MBTA",
                  "agencyTimeZoneOffset":-14400000,
                  "agencyUrl":"http://www.mbta.com",
                  "arrivalDelay":0,
                  "departureDelay":0,
                  "distance":3744.2133081693805,
                  "duration":1020.0,
                  "endTime":1563741540000,
                  "from":{
                     "arrival":1563740520000,
                     "departure":1563740520000,
                     "lat":42.276824,
                     "lon":-71.030448,
                     "name":"North Quincy",
                     "stopCode":"9070098",
                     "stopId":"1:9070098",
                     "stopIndex":0,
                     "stopSequence":1,
                     "vertexType":"TRANSIT"
                  },
                  "headsign":"Quincy Center",
                  "interlineWithPreviousLeg":false,
                  "intermediateStops":[
                     {
                        "arrival":1563740940000,
                        "departure":1563740940000,
                        "lat":42.26884,
                        "lon":-71.018107,
                        "name":"Hancock St @ Woodbine St",
                        "stopCode":"3025",
                        "stopId":"1:3025",
                        "stopIndex":1,
                        "stopSequence":182,
                        "vertexType":"TRANSIT"
                     },
                     {
                        "arrival":1563741060000,
                        "departure":1563741060000,
                        "lat":42.266671,
                        "lon":-71.017924,
                        "name":"Beale St @ Wollaston Branch Library",
                        "stopCode":"9270099",
                        "stopId":"1:9270099",
                        "stopIndex":2,
                        "stopSequence":190,
                        "vertexType":"TRANSIT"
                     },
                     {
                        "arrival":1563741180000,
                        "departure":1563741180000,
                        "lat":42.26502,
                        "lon":-71.019245,
                        "name":"Newport Ave @ Beale St",
                        "stopCode":"9170099",
                        "stopId":"1:9170099",
                        "stopIndex":3,
                        "stopSequence":195,
                        "vertexType":"TRANSIT"
                     }
                  ],
                  "legGeometry":{
                     "length":59,
                     "points":"af`aGhbppLKUEYMSAQBQhGqJdAqAdKiKj@k@dP}ZhIiPdDsGJSNIDAJAF@F@NFzA|AnBdDR\\\\p@f@|@hBtCNJLFN?LARKf@g@pCwCtEaFpYm[xAsAbCwArBcBR]R_@R{@Vw@~BcCNMJSDYBY?e@DWXi@Ze@fJ{Q^k@\\a@tBeBtCmBhAo@XU@A"
                  },
                  "mode":"BUS",
                  "pathway":false,
                  "realTime":false,
                  "rentedBike":false,
                  "route":"Red Line Shuttle",
                  "routeColor":"FFC72C",
                  "routeId":"1:Shuttle005",
                  "routeLongName":"Red Line Shuttle",
                  "routeTextColor":"000000",
                  "routeType":3,
                  "serviceDate":"20190721",
                  "startTime":1563740520000,
                  "steps":[

                  ],
                  "to":{
                     "arrival":1563741540000,
                     "departure":1563742270000,
                     "lat":42.251786,
                     "lon":-71.003876,
                     "name":"Quincy Center",
                     "stopCode":"9070101",
                     "stopId":"1:9070101",
                     "stopIndex":4,
                     "stopSequence":200,
                     "vertexType":"TRANSIT"
                  },
                  "transitLeg":true,
                  "tripId":"1:40525673-20:45-BraintreeNQuincyL1"
               },
               {
                  "agencyTimeZoneOffset":-14400000,
                  "arrivalDelay":0,
                  "departureDelay":0,
                  "distance":124.59100000000001,
                  "duration":110.0,
                  "endTime":1563742380000,
                  "from":{
                     "arrival":1563741540000,
                     "departure":1563742270000,
                     "lat":42.251786,
                     "lon":-71.003876,
                     "name":"Quincy Center",
                     "stopCode":"9070101",
                     "stopId":"1:9070101",
                     "stopIndex":4,
                     "stopSequence":200,
                     "vertexType":"TRANSIT"
                  },
                  "interlineWithPreviousLeg":false,
                  "intermediateStops":[

                  ],
                  "legGeometry":{
                     "length":11,
                     "points":"mh{`Gz}jpLEBb@pBGj@@HBN@FOHSJOHEB"
                  },
                  "mode":"WALK",
                  "pathway":false,
                  "realTime":false,
                  "rentedBike":false,
                  "route":"",
                  "startTime":1563742270000,
                  "steps":[
                     {
                        "absoluteDirection":"NORTHWEST",
                        "area":false,
                        "bogusName":true,
                        "distance":3.313,
                        "elevation":[

                        ],
                        "lat":42.25175378164277,
                        "lon":-71.00397604247063,
                        "relativeDirection":"DEPART",
                        "stayOn":false,
                        "streetName":"sidewalk"
                     },
                     {
                        "absoluteDirection":"SOUTHWEST",
                        "area":false,
                        "bogusName":true,
                        "distance":121.27799999999999,
                        "elevation":[

                        ],
                        "lat":42.2517811,
                        "lon":-71.0039921,
                        "relativeDirection":"LEFT",
                        "stayOn":true,
                        "streetName":"path"
                     }
                  ],
                  "to":{
                     "arrival":1563742380000,
                     "lat":42.251809,
                     "lon":-71.005409,
                     "name":"Quincy Center",
                     "orig":"Quincy Center",
                     "vertexType":"NORMAL"
                  },
                  "transitLeg":false
               }
            ],
            "startTime":1563739241000,
            "tooSloped":false,
            "transfers":1,
            "transitTime":1980,
            "waitingTime":863,
            "walkDistance":124.68498495937342,
            "walkLimitExceeded":false,
            "walkTime":296
         },
         {
            "duration":4944,
            "elevationGained":0.0,
            "elevationLost":0.0,
            "endTime":1563742380000,
            "legs":[
               {
                  "agencyTimeZoneOffset":-14400000,
                  "arrivalDelay":0,
                  "departureDelay":0,
                  "distance":67.29053121698426,
                  "duration":263.0,
                  "endTime":1563737700000,
                  "from":{
                     "departure":1563737437000,
                     "lat":42.352271,
                     "lon":-71.055242,
                     "name":"South Station",
                     "orig":"South Station",
                     "vertexType":"NORMAL"
                  },
                  "interlineWithPreviousLeg":false,
                  "intermediateStops":[

                  ],
                  "legGeometry":{
                     "length":2,
                     "points":"u|naGh~tpLdByA"
                  },
                  "mode":"WALK",
                  "pathway":false,
                  "realTime":false,
                  "rentedBike":false,
                  "route":"",
                  "startTime":1563737437000,
                  "steps":[
                     {
                        "absoluteDirection":"SOUTHEAST",
                        "area":false,
                        "bogusName":false,
                        "distance":67.29053121698426,
                        "elevation":[

                        ],
                        "lat":42.352271,
                        "lon":-71.055242,
                        "relativeDirection":"DEPART",
                        "stayOn":false,
                        "streetName":"Transfer"
                     }
                  ],
                  "to":{
                     "arrival":1563737700000,
                     "departure":1563737700000,
                     "lat":42.351763,
                     "lon":-71.054797,
                     "name":"South Station",
                     "stopId":"1:South Station",
                     "stopIndex":0,
                     "stopSequence":1,
                     "vertexType":"TRANSIT",
                     "zoneId":"CR-zone-1A"
                  },
                  "transitLeg":false
               },
               {
                  "agencyId":"1",
                  "agencyName":"MBTA",
                  "agencyTimeZoneOffset":-14400000,
                  "agencyUrl":"http://www.mbta.com",
                  "alightRule":"coordinateWithDriver",
                  "arrivalDelay":0,
                  "departureDelay":0,
                  "distance":12027.831532914552,
                  "duration":840.0,
                  "endTime":1563738540000,
                  "from":{
                     "arrival":1563737700000,
                     "departure":1563737700000,
                     "lat":42.351763,
                     "lon":-71.054797,
                     "name":"South Station",
                     "stopId":"1:South Station",
                     "stopIndex":0,
                     "stopSequence":1,
                     "vertexType":"TRANSIT",
                     "zoneId":"CR-zone-1A"
                  },
                  "headsign":"Middleborough/Lakeville",
                  "interlineWithPreviousLeg":false,
                  "intermediateStops":[
                     {
                        "arrival":1563738060000,
                        "departure":1563738060000,
                        "lat":42.321079,
                        "lon":-71.052155,
                        "name":"JFK/UMass",
                        "stopId":"1:JFK/UMASS",
                        "stopIndex":1,
                        "stopSequence":2,
                        "vertexType":"TRANSIT",
                        "zoneId":"CR-zone-1A"
                     }
                  ],
                  "legGeometry":{
                     "length":225,
                     "points":"ipnaG`}tpL~An@|Av@zA|@zAz@zA|@zA|@zAz@tAfAzA~@zAz@zA|@zA|@zAz@zA~@|Av@~An@bBVdBJfB?dBAdBEdBEfBCdBEdBEdBCfBDdBBdB@dBCdBMdBWbBYdBYdBIdBIdBGfBGdBGdBCdBCfBAbB]`Bi@|Au@vAeAnAuAlA{AjA{AhAaBfAaBhA_BfAaBhAaBhA_BjA}ArAkAxAaA`Bi@bBYdBQdBO@?bBOdBKdBKfBEdBCdBAfB?dB?dBGdBEdBTdB@dBFdBTbB^`Bf@`Bf@`Bf@`Bf@`Bh@`Bh@`Bf@`Bh@bBb@bB\\bBZdBXbBXbBZdBVdBVbBVbB^bBZdBVdBLdB?dBKdBY`Bi@|Aw@xAaArAmAnAwAnAwAlAwAnAwArAmAnAwAlAyAnAwAnAwAlAwAnAwAlAyAlAwAnAyAlAwAnAyAlAyAlAwAnAyAlAwAlAyAnAwAlAwAnAyAnAwAlAyAlAyAlAyAlAyAlAwAnAyAnAuAnAuApAsAnAwAlAwAnAwAnAyAlAyAlAwAlAyAlAyAnAyAlAwAlA{AnAuArAkAxAgA|As@`Bg@bB_@bBe@`Be@~Ao@|Aw@tAkApAsApAqArAqApAsApAsAnAsAnAwAnAwAlAyAlAyAnAwAlAyAnAuApAsApAuApAsAnAsAnAuAnAwAnAuApAuAnAwAnAuAnAuAnAwApAsAlAyApAsApAqArAqAnAsAnAwApAuAnAuApAsAnAuAnAuAnAwApAuAnAsArAqApAsAnAuAnAuApAuAnAuAnAuAnAuApAuApAuAnAsApAsAnAuAnAwAnAuAnAyApAqArAoAxAaA`Bi@zA}@RM"
                  },
                  "mode":"RAIL",
                  "pathway":false,
                  "realTime":false,
                  "rentedBike":false,
                  "route":"Middleborough/Lakeville Line",
                  "routeColor":"80276C",
                  "routeId":"1:CR-Middleborough",
                  "routeLongName":"Middleborough/Lakeville Line",
                  "routeTextColor":"FFFFFF",
                  "routeType":2,
                  "serviceDate":"20190721",
                  "startTime":1563737700000,
                  "steps":[

                  ],
                  "to":{
                     "arrival":1563738540000,
                     "lat":42.251809,
                     "lon":-71.005409,
                     "name":"Quincy Center",
                     "orig":"Quincy Center",
                     "stopId":"1:Quincy Center",
                     "stopIndex":2,
                     "stopSequence":3,
                     "vertexType":"TRANSIT",
                     "zoneId":"CR-zone-1A-1"
                  },
                  "transitLeg":true,
                  "tripId":"1:CR-Sunday-Spring-19-2009",
                  "tripShortName":"2009"
               }
            ],
            "startTime":1563737436000,
            "tooSloped":false,
            "transfers":0,
            "transitTime":840,
            "waitingTime":3841,
            "walkDistance":0.0,
            "walkLimitExceeded":false,
            "walkTime":263
         }
      ],
      "to":{
         "lat":42.251809,
         "lon":-71.005409,
         "name":"Quincy Center",
         "orig":"Quincy Center",
         "vertexType":"NORMAL"
      }
   },
   "requestParameters":{
      "arriveBy":"true",
      "date":"2019-07-21",
      "format":"json",
      "fromPlace":"South Station::42.352271,-71.055242",
      "locale":"en",
      "mode":"TRAM,SUBWAY,FERRY,RAIL,BUS,WALK",
      "showIntermediateStops":"true",
      "time":"04:53PM",
      "toPlace":"Quincy Center::42.251809,-71.005409",
      "walkReluctance":"5"
   }
}
```